src/components/GridCard/GridCard.js

```diff
@@ -12,7 +19,11 @@ const GridCardComponent = ({className, cardId, url, name, category, price}) => {
             <span>{name}</span>
             <span>{category}</span>
             <span>{`$${price}`}</span>
-            <button>Add to cart</button>
+            <input 
+                className="addToCart" 
+                type="submit" 
+                value="Add to Cart"
+                onClick={() => { addToCart({id: cardId, quantity: parseInt(1), price: price, stock: parseInt(stock)}); }}></input>
```

Comment:
Handling this onClick event on a function on and just calling it here is better imo. 
```suggestion
                onClick={ handleOnClick }></input>
```
and the fn something like this
```
const handleOnClick = () => { // there is no need of adding properties because you are not passing any directly
   // then inside here 
   addToCart({
      id: cardId, 
      quantity: parseInt(1), // *
      price, // as the property has the same name as value you can leave it like this, also if you rename cardId to id 
                  you can do this. 
      stock: parseInt(stock)} // **
)};
```
`*`  I have to ask, why passing the number through parsInt? as you already have a number directly in the property the parseInt is not achieving anything
`* *` does the property comes from a string?

Reply:
The property was being considered a string inside the reducer for some reason and was throwing NaN when computed by the addToCart action. I will consider moving the conversion inside the reducer.

Reply:
I've found the issue with the NaN: I was not getting the stock property from the API call. Oops. Removing parseInt's